providers/base/tests/test_pipewire_utils.py

```diff
@@ -0,0 +1,634 @@
+        # with self.assertLogs("root", level="INFO") as cm:
+        #     wrong_type1 = "videog"
+        #     wrong_type2 = "xxxx"
+        #     self.assertEqual("UNKNOWN TYPE",
+        #                      pt.generate_pw_media_class(wrong_type1,
+        #                                                 "source"))
+        #     self.assertEqual("UNKNOWN TYPE",
+        #                      pt.generate_pw_media_class(wrong_type2,
+        #                                                 "sources"))
+        #     # check log output
+        #     self.assertEqual(cm.output,
+        #                      ["INFO:root:Media type:[{}] is unknown"
+        #                       .format(wrong_type1),
+        #                       "INFO:root:Media type:[{}] is unknown"
+        #                       .format(wrong_type2)])
+
+        # return UNKNOWN CLASS
+        wrong_class1 = "sourceg"
+        wrong_class2 = "sinkf"
+        self.assertEqual("UNKNOWN CLASS",
+                         pt.generate_pw_media_class("video",
+                                                    wrong_class1))
+        self.assertEqual("UNKNOWN CLASS",
+                         pt.generate_pw_media_class("video",
+                                                    wrong_class2))
+
+
+class DetectDeviceTests(unittest.TestCase):
+
+    # Correct Node
+    sink_audio_node = '[{\
+                        "id": 29,\
+                        "type": "PipeWire:Interface:Node",\
+                        "version": 3,\
+                        "permissions": [ "r", "w", "x", "m" ],\
+                        "info": {\
+                          "max-input-ports": 0,\
+                          "max-output-ports": 0,\
+                          "change-mask": [ "input-ports",\
+                                           "output-ports",\
+                                           "state", "props",\
+                                           "params" ],\
+                          "n-input-ports": 0,\
+                          "n-output-ports": 0,\
+                          "state": "suspended",\
+                          "error": null,\
+                          "props": {\
+                            "media.class": "Audio/Sink",\
+                            "factory.name": "support.node.driver",\
+                            "node.name": "Freewheel-Driver",\
+                            "priority.driver": 19000,\
+                            "node.group": "pipewire.freewheel",\
+                            "node.freewheel": true,\
+                            "factory.id": 10,\
+                            "clock.quantum-limit": 8192,\
+                            "node.driver": true,\
+                            "object.id": 29,\
+                            "object.serial": 29\
+                          },\
+                          "params": {\
+                          }\
+                        }\
+                      }]'
```

Comment:
please use multiline strings

Reply:
Change this to suggestion according to code review criteria, feel free to close it.